builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -112,6 +116,8 @@ func (s *StepWatchSource) initStep(state multistep.StateBag) error {
 	s.SourceName = sourceName
 	s.Namespace = namespace
 	s.KubeWatchClient = kubeWatchClient
+	s.timeoutCh = time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second)
```

Comment:
We can replace this by using a context that is created within the Run function. I found this to be a bit confusing at first but now understand its usages given our internal chat. 

```suggestion
```

FYI the timer will start upon initialization.